src/wirecloud/catalogue/search.py

```diff
@@ -0,0 +1,317 @@
+            "vendor_name": {"type": "keyword"},
+            "vendor": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard",
+                "fields": {
+                    "keyword": {"type": "keyword"}
+                }
+            },
+            "name": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard",
+                "fields": {
+                    "keyword": {"type": "keyword"}
+                }
+            },
+            "version": {"type": "keyword"},
+            "version_sortable": {"type": "float"},
+            "description_url": {"type": "keyword"},
+            "type": {"type": "keyword"},
+            "creation_date": {"type": "date", "format": "strict_date_optional_time||epoch_millis"},
+            "public": {"type": "boolean"},
+            "title": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard"
+            },
+            "endpoint_descriptions": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard"
+            },
+            "description": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard"
+            },
+            "image": {"type": "keyword"},
+            "smartphoneimage": {"type": "keyword"},
+            "users": {"type": "keyword"},
+            "groups": {"type": "keyword"},
+            "input_friendcodes": {"type": "keyword"},
+            "output_friendcodes": {"type": "keyword"}
+        }
+    }
+}
+
+
+class ResourceBase(BaseModel):
+    id: str
+    vendor_name: str
+    vendor: str
+    name: str
+    version: str
+    description_url: str
+    type: str
+    creation_date: datetime
+    public: bool
+    title: str
+    description: str
+    image: str
+    smartphoneimage: str
+    input_friendcodes: Any
+    output_friendcodes: Any
+
+
+class ResourceOut(ResourceBase):
+    version_sortable: float
+    endpoint_descriptions: str
+    users: list[str]
+    groups: list[str]
+
+
+class ResourceOutResponse(ResourceBase):
+    others: list[str]
+    uri: str
+
+
+def build_version_sortable(version: str):
+    code = 0
+    ver = Version(version)
+
+    code += ver.version[0] * 1000 * 1000
+    code += ver.version[1] * 1000 if len(ver.version) > 1 else 0
+    code += ver.version[2] if len(ver.version) > 2 else 0
+
+    prerelease = ver.prerelease
+    if prerelease is None:
+        code += .999
+    elif prerelease[0] == "a":
+        code += prerelease[1] / 1000
+    elif prerelease[0] == "b":
+        code += (100 + prerelease[1]) / 1000
+    else:
+        code += (200 + prerelease[1]) / 1000
+
+    return code
+
+def clean_resource_out(hit: dict, request: Request) -> ResourceOutResponse:
```

Comment:
Dos espacios entre funciones.
```suggestion

def clean_resource_out(hit: dict, request: Request) -> ResourceOutResponse:
```